src/components/Subthoughts.js

```diff
@@ -226,6 +228,26 @@ export const Subthoughts = connect(({ cursorBeforeEdit, cursor, contextViews, th
           : showContexts ? getContextsSortedAndRanked(/* subthought() || */headValue(thoughtsRanked))
             : getThoughtsRanked(contextBinding || thoughtsRanked)
 
+      // Ensure that editable newTought is visible.
+      const editIndex = (cursor && children) ? children.findIndex(child => {
+        const childPath = getChildPath(child, thoughtsRanked, showContexts)
+        const isEditingPath = subsetThoughts(cursor, childPath)
+        return equalPath(cursor, childPath) || isEditingPath
+      }) : 0
+      const filteredChildren = children.filter(child => {
+        const value = showContexts ? head(child.context) : child.value
+        return showHiddenThoughts ||
+          (!isFunction(value) && !meta(pathToContext(unroot(thoughtsRanked)).concat(value)).hidden)
+      })
+      const paginationSize = isMobile ? PAGINATION_SIZE_MOBILE : PAGINATION_SIZE_DESKTOP
+      if (editIndex > paginationSize * page - 1) {
+        setPage(page + 1)
```

Comment:
The pagination logic should not be executed if `!show`, since the subthoughts would not be rendered anyway.

Reply:
done

Reply:
It looks like this was done partially. `paginationSize` and `proposedPageSize` are still calculated regardless. It would be nice to avoid all of the pagination logic when `!show`.

Reply:
There are no extra loops operating. adding extra logic to avoid just simple + - calculations will end having a lot of extra logic and overall page performance will be even slower than just having few extra variables that are not used in !show scenarios.

Reply:
I believe the extra logic is basically just `if (show) { ... }`. That would technically be faster than the calculations. Is there other logic I am not aware of?